src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1240,11 +1242,11 @@ private static TokenValidationResult ReadToken(string token, TokenValidationPara
             {
                 var securityToken = validationParameters.TokenReader(token, validationParameters);
                 if (securityToken == null)
-                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10510, token)));
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10510, JwtTokenUtilities.PrepareSecurityArtifact(token))));
```

Comment:
I would prefer we followed the pattern used for PII. Something like, `LogHelper.MarkSecurityArtifact`. This has a couple advantages in my mind
1. It reuses a pattern which is already established in code so it makes it easier to understand what's going on.
2. It allows log level to be taken into account before performing any string operations. We can win from a performance standpoint if we're only doing these string operations (or regEx searches) when we actually go to log. If there's a log informational with a token. Code as written would still do the string manipulation even if the caller had only enabled error logs.